packages/next/server/base-server.ts

```diff
@@ -1040,8 +1029,33 @@ export default abstract class Server<ServerOptions extends Options = Options> {
 
       if (hasFallback || staticPaths?.includes(resolvedUrlPathname)) {
         isSSG = true
+      } else if (!this.renderOpts.dev) {
+        const manifest = this.getPrerenderManifest()
+        isSSG =
+          isSSG || !!manifest.routes[pathname === '/index' ? '/' : pathname]
+      }
+    }
+
+    // Toggle whether or not this is a Data request
+    let isDataReq =
+      !!(
+        query.__nextDataReq ||
+        (req.headers['x-nextjs-data'] &&
+          (this.serverOptions as any).webServerConfig)
+      ) &&
+      (isSSG || hasServerProps)
+
+    if (isAppPath && req.headers['__rsc__']) {
+      res.setHeader('content-type', 'application/octet-stream')
```

Comment:
It seems content-type is set here regardless of what comes out of app-render, is that intentional?

Reply:
Updated